packages/react-reconciler/src/ReactProfileTimer.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {Fiber} from './ReactFiber';
+import type {StackCursor} from './ReactFiberStack';
+
+import warning from 'fbjs/lib/warning';
+import ReactFiberStack from './ReactFiberStack';
+
+const hasNativePerformanceNow =
+  typeof performance === 'object' && typeof performance.now === 'function';
+
+let now;
+if (hasNativePerformanceNow) {
+  now = function() {
+    return performance.now();
+  };
+} else {
+  now = function() {
+    return Date.now();
+  };
+}
+
+/**
+ * The "actual" render time is total time required to render the descendants of a Profiler component.
+ * This time is stored as a stack, since Profilers can be nested.
+ * This time is started during the "begin" phase and stopped during the "complete" phase.
+ * It is paused (and accumulated) in the event of an interruption or an aborted render.
+ */
+
+const {checkThatStackIsEmpty, createCursor, push, pop} = ReactFiberStack();
```

Comment:
We should never create new fiber stacks anywhere except once.

We should reuse the one from here:
https://github.com/facebook/react/blob/b548b3cd640dbd515f5d67dafc0216bb7ee0d796/packages/react-reconciler/src/ReactFiberScheduler.js#L161-L164

Reply:
Interesting.

I avoided doing this initially because:
* The existing stack seems completely context-based, so I wasn't sure if it was appropriate to intermingle timing values. (I was also concerned about adding complexity in terms of push/pop timing when adding more hooks into the existing stack.)
* Re-using the existing stack cursor makes the profiler functions more awkward to use and pass around, particularly given the feature flag constraints.
* A stack seemed like a pretty lightweight structure.

I can make this change though, if you think it's best.